### DISCUSSION TITLE
Why my tiles flying above the map, even with existing mapbox terrain?

### DISCUSSION THREAD

**Discussion by bolatuly** [visgl/deck.gl#5783] [Q&A]
Hi guys,

Thank you for this great repo. I use it for displaying 3d tiles in mapbox.

As you know probably know, in mapbox the basemap is always rendered at altitude=0. Since my 3d tiles use real-world elevation data, they were flying above the map. I used this code to push the tile's geometries down to 0:

```javascript
// find cartographic center of bounding volume
const cartographic = Cesium.Cartographic.fromCartesian(
        new Cesium.Cartesian3(sphere[0], sphere[1], sphere[2])
)
// find relative coordinates on the ground
const ground = Cesium.Cartesian3.fromRadians(cartographic.longitude, cartographic.latitude, 0);

// find offset
const offset = {
        x: ground.x - sphere[0],
        y: ground.y - sphere[1],
        z: ground.z - sphere[2]
    };

// then set this array to root's transform in tileset.json 
/*
[
        1.0, 
        0.0, 
        0.0, 
        0.0, 
        0.0, 
        1.0, 
        0.0, 
        0.0, 
        0.0, 
        0.0, 
        1.0, 
        0.0, 
        offset.x, 
        offset.y, 
        offset.z,
        1.0
    ];
*/
```
Everything works just fine for mapbox without terrain.

Then I've found that mapbox released a new major version v2 (with terrain). And I thought I no longer need the code above since mapbox now has an elevation and my tiles won't fly above the map. And to my surprise, tiles still flying above the map if I don't transform them. I'm curious why.

Do you have any ideas? Can provide sample code if needed.

I'm not sure it's a bug, that's why I'm writing it here.

**Comment by Pessimistress** [comment]
deck.gl does not support Mapbox terrain at the moment. You can follow the discussion in https://github.com/visgl/deck.gl/issues/5662.

**Reply by bolatuly** [reply]
Could you suggest any hack how to push geometries down or update viewport as proposed [here](https://github.com/visgl/deck.gl/issues/5662#issuecomment-823967148) and [here](https://github.com/visgl/deck.gl/issues/5662#issuecomment-842893008)? I’m using deck.gl’s `MapboxLayer`, implementation of mapbox’s custom layer. Pure js, without any react dependencies.
